packages/state-transition/src/block/processConsolidationRequest.ts

```diff
@@ -30,11 +29,11 @@ export function processConsolidationRequest(
   }
 
   // Verify that source != target, so a consolidation cannot be used as an exit.
-  if (sourceIndex === targetIndex){
+  if (sourceIndex === targetIndex) {
     return;
   }
 
-  const sourceValidator = state.validators.getReadonly(sourceIndex);
+  const sourceValidator = state.validators.get(sourceIndex);
```

Comment:
what necessitated this change?

Reply:
need to modify sourceValidator as below: ` sourceValidator.exitEpoch = exitEpoch;`